golem_messages/message/tasks.py

```diff
@@ -399,12 +389,17 @@ def verify_ethsig(
         associated with `requestor_ethereum_public_key`
 
         :param msg_hash: maybe optionally provided to skip generation
-                         of the message hash during the verification
+                         of the message hash during the verification.
+                         If not provided, the hash is generated from nested
+                         WantToComputeTask.
         :return: `True` if the signature is correct.
         :raises: `exceptions.InvalidSignature` if the signature is corrupted
         """
+        if msg_hash is None and self.want_to_compute_task is not None:
```

Comment:
again, I believe ```
self.want_to_compute_task._verify_signature(
self.ethsig,
decode_hex(self.requestor_ethereum_public_key)`,
msg_hash)
```

should be enough + more readable

Reply:
:+1: done